api/pkg/apis/v1alpha1/managers/configs/configs-manager.go

```diff
@@ -58,7 +58,8 @@ func (s *ConfigsManager) Init(context *contexts.VendorContext, cfg managers.Mana
 	return nil
 }
 func (s *ConfigsManager) Get(object string, field string, overlays []string, localContext interface{}) (interface{}, error) {
-	if strings.Index(object, ":") > 0 {
+	log.Debugf(" M (Config): Get %v, config provider size %d", object, len(s.ConfigProviders))
+	if strings.Index(object, ":") > 0 && len(s.ConfigProviders) > 1 {
```

Comment:
We have a syntax conflict here. Initially, we support <provider>:<key> format so that a user can select reading configurations from a specific configuration provider when multiple providers are configured. Now, as we want to support <key>:<version> syntax, we need either: 1) anticipate there could be three segments; 2) use a different syntax to identify the provider, such as <provider>::<key>:<value>.

Reply:
For the current solution , for one provider scenario, it covers both reference and provider reference, just store/read data and use "provider-xxx" as key. And I do think we need three segments, will allow `len(parts) == 2 or ==3`. What do you think.

Reply:
No so true for scenario if customer use `"provider:abc"` to set and `"abc"` to read. I think it is necessary to use a different seperator for provider `"::"`.